lib/collection/src/collection_manager/collection_updater.rs

```diff
@@ -44,6 +44,12 @@ impl CollectionUpdater {
     ) -> CollectionResult<usize> {
         // Allow only one update at a time, ensure no data races between segments.
         // let _lock = self.update_lock.lock().unwrap();
+        let scroll_lock = segments.read().scroll_read_lock.clone();
+
+        // Use block_in_place here to avoid blocking the current async executor
+        let _scroll_lock =
+            tokio::task::block_in_place(|| tokio::sync::RwLock::write_owned(scroll_lock));
```

Comment:
This returns a future, does it mean that it will not make the update wait until the scroll write lock gets acquired?

Reply:
Thanks! I only half-committed it seems. Bumped it just now in https://github.com/qdrant/qdrant/pull/6289/commits/3b65d006b72994f9294b712381c9fcad858f5035.